Zend/zend_execute_API.c

```diff
@@ -650,6 +650,10 @@ int zend_call_function(zend_fcall_info *fci, zend_fcall_info_cache *fci_cache) /
 
 	ZVAL_UNDEF(fci->retval);
 
+	if (EG(timed_out)) {
+		zend_timeout();
+	}
```

Comment:
It would make more sense to me to check this condition after the function call, rather than before it.

Reply:
I had to think about this a bit, but it seems to make sense...

Reply:
OK, I think it's better now...